### ISSUE TITLE
Cannot translate index entries

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#976]
When definining index entries:

```
.. index::
    single: Mailing List
    pair: Newsletter; Recipients List
```

The index terms "Mailing List", "Newsletter" and "Recipients List" - or at least "Newsletter; Recipients List" as a single token - are not extracted by gettext and can thus not get translated.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/976
- Originally reported by: Xavier Perseguers
- Originally created at: 2012-07-25T09:18:18.230

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-01-05 16:32:30+00:00_

Please review pull request #104

**Comment by shimizukawa** [comment]
_From Xavier Perseguers on 2013-01-05 19:30:07+00:00_

(Reply via xav...[causal](https://bitbucket.org/causal).ch):

Hi,

By reading you have a special processing for "pair" of index terms. However sphinx allows "triple" (IIRC) or something similar as well and I guess the corresponding implementation is missing it altogether.

Thanks 

Xavier Perseguers
Sent from my iPhone

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-01-06 01:24:01+00:00_

[Xavier Perseguers](https://bitbucket.org/xperseguers) Yes, "pair" index term have "built-in translated" keywords then the special processing part remove "built-in translated" strings from `make gettext` output.

Of cause all index types ("single", "pair", "triple", "see", "seealso", "module", "keyword", "operator", "object", "exception", "statement", "builtin") become translatable by pull request #104.

This functionalities are tested with:
- https://bitbucket.org/birkenfeld/sphinx/pull-request/104#chg-tests/root/i18n/index_entries.txt
- https://bitbucket.org/birkenfeld/sphinx/pull-request/104#Ltests/test_build_gettext.pyT101

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-01-06 01:55:56+00:00_

Closes #976: Fix gettext does not extract index entries.

→ <<cset b17c588b0d9c61ba71d299c91c1ac06684f9ebb0>>
